### PR TITLE
Added tests for the vault module.

### PR DESCRIPTION
**Pull request by parthjindal-ibm**
Adding unit tests for the vault module.
Testing all methods and asserting return values and verifying method calls.

### CODE REVIEW

File: internal/vault/src/test/java/com/instana/vault/VaultServiceConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    private VaultServiceConfig config;
+
+    @BeforeEach
+    public void setUp() {
+        config = new VaultServiceConfig();
+    }
+
+    @Test
+    public void testGetConnectionURL() throws Exception {
+        String json = "{\"connection_url\": \"http://localhost:8200\"}";
```
Comment (debnathbikram) [inline internal/vault/src/test/java/com/instana/vault/VaultServiceConfigTest.java:22]: Please create a dummy json file and use it to convert to `VaultServiceConfig`.
Applicable to all the test cases.

**Review by debnathbikram — COMMENTED** [review]
One comment rest looks fine

File: internal/vault/src/test/java/com/instana/vault/VaultServiceConfigTest.java (hunk 1 above)
Comment (parthjindal-ibm) [reply internal/vault/src/test/java/com/instana/vault/VaultServiceConfigTest.java]: I have applied the suggested changes @BikramDebnath1, please have a look.

**Review by debnathbikram — APPROVED** [review]
LGTM

**Pull request merged**